crates/engine/tree/src/tree/root/sync_task.rs

```diff
@@ -0,0 +1,82 @@
+//! Sync state root task implementation.
+
+use super::{StateRootConfig, StateRootHandle, StateRootResult};
+use reth_trie::updates::TrieUpdates;
+use revm_primitives::{EvmState, B256};
+use std::sync::mpsc::{self, Receiver, RecvError};
+use tracing::debug;
+
+/// Wrapper for std channel receiver to maintain compatibility with `UnboundedReceiverStream`
+#[allow(dead_code)]
+pub(crate) struct StdReceiverStream {
+    rx: Receiver<EvmState>,
+}
+
+#[allow(dead_code)]
+impl StdReceiverStream {
+    pub(crate) const fn new(rx: Receiver<EvmState>) -> Self {
+        Self { rx }
+    }
+
+    pub(crate) fn recv(&self) -> Result<EvmState, RecvError> {
+        self.rx.recv()
+    }
+}
+
+/// Synchronous implementation of the state root task
+#[allow(dead_code)]
+pub(crate) struct StateRootSyncTask<Factory> {
+    state_stream: StdReceiverStream,
+    config: StateRootConfig<Factory>,
+}
+
+#[allow(dead_code)]
+impl<Factory> StateRootSyncTask<Factory>
+where
+    Factory: Send + 'static,
+{
+    pub(crate) const fn new(
+        config: StateRootConfig<Factory>,
+        state_stream: StdReceiverStream,
+    ) -> Self {
+        Self { config, state_stream }
+    }
+
+    pub(crate) fn spawn(self) -> StateRootHandle {
+        let (tx, rx) = mpsc::channel();
+        rayon::spawn(move || {
+            debug!(target: "engine::tree", "Starting sync state root task");
+            let result = self.run();
+            let _ = tx.send(result);
+        });
```

Comment:
I think rayon is a bit problematic here because this does a lot of IO.

since we're always using this, we could use a dedicated background thread for this that loops/recv `StateRootSyncTask`s

similar to the existing persistence task

Reply:
indeed, this is what we had initially https://github.com/paradigmxyz/reth/pull/12335/commits/891edaa0af565d8dca19fbc0196e5be66f8a5e6e recovered, also switched from `mpsc::channel` to `mpsc::sync_channel(1)`